cfme/intelligence/chargeback/rates.py

```diff
@@ -73,14 +77,51 @@ def sub_row_id(self):
                 parent=self)
             return int(attr.rsplit('_', 1)[-1])
 
-        per_time = Select(id=ParametrizedString('per_time_{@row_id}'))
-        per_unit = Select(id=ParametrizedString('per_unit_{@row_id}'))
-        start = Input(id=ParametrizedString('start_{@row_id}_{@sub_row_id}'))
-        finish = Input(id=ParametrizedString('finish_{@row_id}_{@sub_row_id}'))
-        fixed_rate = Input(id=ParametrizedString('fixed_rate_{@row_id}_{@sub_row_id}'))
-        variable_rate = Input(id=ParametrizedString('variable_rate_{@row_id}_{@sub_row_id}'))
-        action_add = Button(title='Add a new tier')
-        action_delete = Button(title='Remove the tier')
+        @ParametrizedView.nested
+        class sub_rows(ParametrizedView):  # noqa
+            PARAMETERS = ('index',)
+            ROOT = ParametrizedLocator(
+                '../tr[./td/input[starts-with(@id, "fixed_rate_{@row_id}_{index}")]]')
+
+            def __locator__(self):
+                return self.browser.element(self.ROOT, parent=self.parent)
+
+            @cached_property
+            def row_id(self):
+                return self.parent.row_id
+
+            def fill(self, values):
+                if not self.is_displayed:
+                    self.__class__(index=self.context['index'] - 1).action_add.click()
+                return ParametrizedView.fill(self, values)
+
+            per_time = Select(id=ParametrizedString('per_time_{@row_id}'))
+            per_unit = Select(id=ParametrizedString('per_unit_{@row_id}'))
+            start = Input(id=ParametrizedString('start_{@row_id}_{index}'))
+            finish = Input(id=ParametrizedString('finish_{@row_id}_{index}'))
+            fixed_rate = Input(id=ParametrizedString('fixed_rate_{@row_id}_{index}'))
+            variable_rate = Input(id=ParametrizedString('variable_rate_{@row_id}_{index}'))
+            action_add = Button(title='Add a new tier')
+            action_delete = Button(title='Remove the tier')
+
+        def fill(self, values):
+            """If we want multiple tiers per metric, we can map the values as list (or tuple)
+            and then it'll add a new tiers accordingly (if no such sub row exists).
+            In case that we didn't provide some sub row and it exists, we delete it
+            (since there are some cases that such sub rows exist by default)."""
+            if not isinstance(values, (list, tuple)):
+                values = [values]
+            sub_row_id = 0
+            while True:
+                sub_row = self.sub_rows(index=sub_row_id)
+                if sub_row_id < len(values):
```

Comment:
This is not the logic. I would like to check if there are also sub_rows that should be deleted. also, if you specified sub_rows that should be added you will break the loop before it and you will miss them. Eg: if you provide `values=[{'per_time': 'daily'. 'fixed_rate': 1, 'variable_rate': 2}, {'per_time': 'daily'. 'fixed_rate': 3, 'variable_rate': 4}] ` and there is only 1 sub_row, you should add a new one, but since you iterate over self.sub_rows, you will break the loop and miss the second.